### ISSUE TITLE
Output overwritten when there's no newline

### ISSUE THREAD

**Issue by jaraco** [xonsh/xonsh#2625]
Originally [reported here](https://gitter.im/xonsh/xonsh?at=5adf8cc7012ff2bf681ec37e), when using curl to download content without a terminating newline, the output is overwritten by the prompt.

To replicate, create a server thus:

```
$ open('no-newline.json', 'w').write('{"this doc": "has no new line"}')
31
$ open('no-newline.json').read()
'{"this doc": "has no new line"}'
$ python -m http.server
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) …
```

Then in a xonsh, request the doc:

```
$ curl http://localhost:8000/no-newline.json
$ this doc": "has no new line"}
```

The cursor remains on that last line in pos 3.

**Comment by scopatz** [comment]
[As discussed on gitter,](https://gitter.im/xonsh/xonsh?at=5adf92737c3a01610d1874af) this is due to curl being added as `always_false` to threadable predictors for some reason.

**Comment by scopatz** [comment]
Thanks for reporting @jaraco!

**Comment by scopatz** [comment]
So interstingly, curl contains all of `(b'isatty', b'tcgetattr', b'tcsetattr')`, which is why it is getting set to always_false. I think this is a mistake of the heuristic, though.

**Comment by scopatz** [comment]
potential fix in #2626
